src/ZoneServer/Buffs/Handlers/campfire_Buff.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Melia.Shared.Tos.Const;
+using Melia.Zone.Buffs.Base;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters;
+
+namespace Melia.Zone.Buffs.Handlers
+{
+	/// <summary>
+	/// Handler for the campfire Buff, which increases the target's HP/SP and Stamina recovery
+	/// crit rate.
+	/// </summary>
+	[BuffHandler(BuffId.campfire_Buff)]
+	public class campfire_Buff : BuffHandler
+	{
+		private const string VarName = "Melia.StaminaRecovery";
+		private Dictionary<ICombatEntity, CancellationTokenSource> _cancellationTokenSource = new Dictionary<ICombatEntity, CancellationTokenSource>();
```

Comment:
Oh, you don't want to do that. Storing characters on long-living objects like this could very easily lead to "memory leaks" (not really, but fundamentally similar) when the characters never get removed for some reason. It's also not compatible with relogging and changing maps, which makes it a dangerous pattern that others might copy.

I realize that Bonfire can only be used while sitting, and that it's removed upon logging out by default, but if some user came up with the idea to, for example, make that buff last a while after you got back up, and they changed the end condition and the save value in the data, we'd be in for a bad time.

Reply:
What would you suggest then?
I can't think in any solution atm...